core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1489,7 +1493,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   private[spark] def killAndReplaceExecutor(executorId: String): Boolean = {
     schedulerBackend match {
       case b: CoarseGrainedSchedulerBackend =>
-        b.killExecutors(Seq(executorId), replace = true)
+        b.killExecutors(Seq(executorId), replace = true, true)
```

Comment:
this should be false. We trigger this from a heartbeat. Also please name the boolean parameters `force = false` here and other places.